packages/cli/demo/cat.js

```diff
@@ -585,6 +644,138 @@ const chatComponent = ($parent, powers, { dismissChat }) => {
   return { focusChat, blurChat };
 };
 
+const evalComponent = ($parent, powers, { dismissEval }) => {
+  const $frame = $parent.querySelector('#eval-frame');
+  const $source = $parent.querySelector('#eval-source');
+  const $endOfEndowments = $parent.querySelector('#eval-endowments');
+  const $addEndowment = $parent.querySelector('#eval-add-endowment');
+  const $resultName = $parent.querySelector('#eval-result-name');
+  const $submit = $parent.querySelector('#eval-submit-button');
+  const $discard = $parent.querySelector('#eval-discard-button');
+  let $error = $parent.querySelector('#eval-error');
+  const $endowments = new Map();
+
+  const clearEval = () => {
+    $source.value = '';
+    $resultName.value = '';
+    for (const $endowment of $endowments.values()) {
+      $endowment.remove();
+    }
+    $endowments.clear();
+    const $newError = document.createTextNode('');
+    $error.replaceWith($newError);
+    $error = $newError;
+    dismissEval();
+  };
+
+  $discard.addEventListener('click', () => {
+    clearEval();
+  });
+
+  const handleRemoveEndowment = event => {
+    const { target } = event;
+    const id = target.getAttribute('id');
+    const $endowment = $endowments.get(id);
+    if ($endowment === undefined) {
+      throw new Error(`Endowment does not exist for id ${id}`);
+    }
+    $endowments.delete(id);
+    $endowment.remove();
+  };
+
+  const handleAddEndowment = () => {
+    const $endowment = document.createElement('p');
+    const codeNameId = nextId();
+    const petNameId = nextId();
+    const removeId = nextId();
+
+    $endowment.innerHTML = `
+      <label for="${codeNameId}">
+        Name in source:
+        <input id="${codeNameId}" type="text" class="big half-wide code-name">
+      </label>
+      <label for="${petNameId}">
+        Pet name:
+        <input id="${petNameId}" type="text" class="big half-wide pet-name">
+      </label>
+      <button id="${removeId}">Remove</button>
+    `;
+
+    const $remove = $endowment.querySelector(`#${removeId}`);
+    $remove.addEventListener('click', handleRemoveEndowment);
+    $endOfEndowments.parentElement.insertBefore($endowment, $endOfEndowments);
+    $endowments.set(removeId, $endowment);
+
+    const $codeName = $endowment.querySelector('.code-name');
+    $codeName.focus();
+  };
+
+  $addEndowment.addEventListener('click', handleAddEndowment);
+
+  const handleEval = () => {
+    const source = $source.value;
+    const workerName = 'MAIN';
+    const names = Array.from($endowments.values(), $endowment => {
+      const $codeName = $endowment.querySelector('.code-name');
+      const $petName = $endowment.querySelector('.pet-name');
+      return {
+        petName: $petName.value,
+        codeName: $codeName.value,
+      };
+    });
+    const codeNames = names.map(({ codeName }) => codeName);
+    const petNames = names.map(({ petName }) => petName);
+    const resultName = $resultName.value.trim() || undefined;
+    E(powers)
```

Comment:
```suggestion
    const resultName = $resultName.value.trim() || undefined;
    console.log('eval', {
      source,
      workerName,
      codeNames,
      petNames,
      resultName,
    });

    E(powers)
```

Reply:
Maybe we should log the result as well?

Reply:
This entrains into scope the generation of a web representation of passable values, which we would need for other reasons, like revealing a representation of a pet-named value. It might be okay to just transition to a subsequent modal view.

Reply:
I added a modal “value” viewer.

<img width="1008" alt="Screenshot 2024-05-03 at 5 10 35 PM" src="https://github.com/endojs/endo/assets/60294/272af81c-4651-4d19-88de-4128fdf9903b">

<img width="1008" alt="Screenshot 2024-05-03 at 5 10 40 PM" src="https://github.com/endojs/endo/assets/60294/8f1ce7ac-e11f-44db-8034-da4ad3c87da3">

Reply:
Now with responsive square brackets and curly braces made possibly by crimes.

<img width="1008" alt="image" src="https://github.com/endojs/endo/assets/60294/51c77a95-b41c-4d20-ab3c-1780fd33a3cb">

Reply:
Why not a plain text rendering with q?

Reply:
RIP. I can certainly make it look like `JSON.stringify` (`q`), and copy-paste nicely, but would want to do better for other passables like bigint, promise, remotable, and tagged. Using a DOM representation allows for responsive layout. Named remotables might be labeled by their pet name. It’s conceivable that CSS could be tricked into pivoting long lines from columns to rows with indentation if they wrap, or just toggle on clicking the brace/bracket. Building out of DOM elements leaves the door open for more interactivity in future changes.

Reply:
Reduced fanciness.

<img width="1042" alt="Screenshot 2024-05-06 at 7 31 10 PM" src="https://github.com/endojs/endo/assets/60294/566c18c8-fe8f-4d29-9637-d07dffe5fe33">

<img width="1774" alt="Screenshot 2024-05-06 at 7 31 24 PM" src="https://github.com/endojs/endo/assets/60294/57419542-6ec3-4b34-bf27-40ddfb337189">

Reply:
> Named remotables might be labeled by their pet name.

That's a _very_ interesting possibility.
IOU an issue.